skimage/io/_plugins/matplotlib_plugin.py

```diff
@@ -1,12 +1,154 @@
+from collections import namedtuple
+import numpy as np
+import warnings
 import matplotlib.pyplot as plt
+from skimage.util import dtype as dtypes
 
 
-def imshow(*args, **kwargs):
-    if plt.gca().has_data():
-        plt.figure()
+_default_colormap = 'gray'
+_nonstandard_colormap = 'cubehelix'
+_diverging_colormap = 'RdBu'
+
+
+ImageProperties = namedtuple('ImageProperties',
+                             ['signed', 'out_of_range_float',
+                              'low_dynamic_range', 'unsupported_dtype'])
+
+
+def _get_image_properties(image):
+    """Determine nonstandard properties of an input image.
+
+    Parameters
+    ----------
+    image : array
+        The input image.
+
+    Returns
+    -------
+    ip : ImageProperties named tuple
+        The properties of the image:
+
+        - signed: whether the image has negative values.
+        - out_of_range_float: if the image has floating point data
+          outside of [-1, 1].
+        - low_dynamic_range: if the image is in the standard image
+          range (e.g. [0, 1] for a floating point image) but its
+          dynamic range would be too small to display with standard
+          image ranges.
+        - unsupported_dtype: if the image data type is not a
+          standard skimage type, e.g. ``numpy.uint64``.
+    """
+    immin, immax = np.min(image), np.max(image)
+    imtype = image.dtype.type
+    try:
+        lo, hi = dtypes.dtype_range[imtype]
+    except KeyError:
+        lo, hi = immin, immax
+
+    signed = immin < 0
+    out_of_range_float = (np.issubdtype(image.dtype, np.float) and
+                          (immin < lo or immax > hi))
+    low_dynamic_range = (immin != immax and
+                         (float(immax - immin) / (hi - lo)) < (1. / 255))
+    unsupported_dtype = image.dtype not in dtypes._supported_types
+
+    ip = ImageProperties(signed, out_of_range_float,
+                         low_dynamic_range, unsupported_dtype)
+    return ip
```

Comment:
Maybe just return the ImageProperties directly?  The variable seems superfluous.